src/library/scala/util/Either.scala

```diff
@@ -621,11 +621,26 @@ object Either {
      *  Right(12).left.filter(_ > 10) // None
      *  }}}
      */
+    @deprecated("Renamed filterToOption in order to more accurately reflect return type.", "2.13.0")
```

Comment:
In general, I think it's better for the deprecation message to tell you what to do, rather than what changed. e.g. something like

```suggestion
    @deprecated("Use `filterToOption`, which more accurately reflects the return type", "2.13.0")
```

Reply:
Thanks for this